owa/app/js/components/manage_patient_identifier_sources/existingIdentifierSources.jsx

```diff
@@ -10,15 +16,24 @@ export default class ExistingIdentifierSources extends Component {
     this.state = {
       modal: false,
       modalContent: '',
-      identifierSources: [{
-        identifierType: ['OpenMRS ID'],
-        sourceType: ['Local Identifier Generator'],
-        sourceName: ['New Source']
-      }]
+      identifierSource: '',
+      identifierSources: [],
+      saveButtonDisplay: 'block'
     };
 
     this.toggle = this.toggle.bind(this);
     this.handleClick = this.handleClick.bind(this);
+    this.handleUpdate = this.handleUpdate.bind(this);
+  }
+
+  componentDidMount(){
+    this.fetchIdentifierSources();
+  }
+
+  componentWillReceiveProps(nextProps){
+    let existingSources = this.state.identifierSources;
+    existingSources.push(nextProps["newIdentifierSource"]);
```

Comment:
You might consider using `Object.Assign` or the `spread` to add the `newIdentifierSource` to what you already have in your state. It's going to make the code look cleaner.

Reply:
fixed